src/java/org/apache/cassandra/db/ColumnIndex.java

```diff
@@ -123,14 +123,20 @@ public void buildRowIndex(UnfilteredRowIterator iterator) throws IOException
     private void writePartitionHeader(UnfilteredRowIterator iterator) throws IOException
     {
         ByteBufferUtil.writeWithShortLength(iterator.partitionKey().getKey(), writer);
+
+        long partitionDeletionPosition = writer.position();
         DeletionTime.serializer.serialize(iterator.partitionLevelDeletion(), writer);
+        if (!observers.isEmpty())
+            observers.forEach((o) -> o.partitionLevelDeletion(iterator.partitionLevelDeletion(), partitionDeletionPosition));
```

Comment:
Nit: we can use the single argument of the lambda without parenthesis, here and in the other two calls:
```suggestion
            observers.forEach(o -> o.partitionLevelDeletion(iterator.partitionLevelDeletion(), partitionDeletionPosition));
```